nexus/db-model/src/certificate.rs

```diff
@@ -82,6 +82,16 @@ impl TryFrom<Certificate> for views::Certificate {
         Ok(Self {
             identity: cert.identity(),
             service: cert.service.try_into()?,
+            cert: String::from_utf8(cert.cert)
+                .map_err(|err| {
+                    Error::InternalError {
+                        internal_message: format!(
+                            "Failed to construct string from stored certificate: {}",
+                            err
+                        )
+                    }
+                }
+            )?,
```

Comment:
Worth having a comment about why this conversion is expected to always succeed. As far as I can tell, we should expect this conversion to work 100% of the time because it comes in as a `String`. 

https://github.com/oxidecomputer/omicron/blob/5176bb3c0398488d0c9669fa049e85b33fbacd54/nexus/types/src/external_api/params.rs#L811-L823

I figure the 500 is appropriate. Maybe be more concrete in the error, like: `cert was not valid UTF-8`. Though the `err` you stick in there should say that too. In general using this constructor function is slightly preferred over the struct constructor, but it doesn't matter that much, and it won't be much less noisy.

https://github.com/oxidecomputer/omicron/blob/5176bb3c0398488d0c9669fa049e85b33fbacd54/common/src/api/external/error.rs#L211-L219

Reply:
Updated the error here and added a comment on why I expect the conversion not to fail. I think your more direct error explaining that the certificate is not valid utf-8 is better. The 500 felt off to me initially too. But It is probably correct as it it should be something that the control plane did that got us in to the error case.